src/main/java/gui/view/MainPageController.java

```diff
@@ -0,0 +1,162 @@
+            HBox task = new HBox(taskName, deadline);
+            taskListView.getItems().add(task);
+        }
+        taskListView.refresh();
+
+    }
+
+    /**
+     * Note: removes visibility of all previously rendered labels in the list view before flushing it out.
+     * @param map a mapping of name and time for today's relevant items
+     */
+    private void updateEventListView(Map<String, List<String>> map) {
+        eventListView.getItems().forEach(i -> i.setVisible(false));
+        eventListView.getItems().clear();
+        for (Map.Entry<String, List<String>> eventInfo : map.entrySet()) {
+            Label eventName = new Label(eventInfo.getKey());
+            String startTime = "";
+            String endTime = "";
+            if (!(eventInfo.getValue() == null)) {
+                startTime = eventInfo.getValue().get(0);
+                endTime = eventInfo.getValue().get(1);
+            }
+            Label duration = new Label(startTime + " - " + endTime);
+            eventName.setFont(new Font(labelFontSize));
+            duration.setFont(new Font(labelFontSize));
+            eventName.setMinWidth(580);
+            eventName.setMaxWidth(580);
+
+            HBox task = new HBox(eventName, duration);
+            eventListView.getItems().add(task);
+        }
+        eventListView.refresh();
+
+    }
+
+
+    /**
+     * Method listening to updates from the {@link MainPageViewModel}.
+     * @param infos the lists of updated infos
+     */
+    private void acceptReaders(MainPageViewModel.Infos infos) {
+        updateTaskListView(formatTaskInfos(infos.taskInfos));
+        updateEventListView(formatEventInfos(infos.eventInfos));
+    }
+
+    private Map<String, String> formatTaskInfos(List<TaskInfo> taskInfo) {
+        Map<String, String> taskMap = new HashMap<>();
+        if (taskInfo.isEmpty()){
+            taskMap.put("No Scheduled Tasks", null);
+        }
+        for (TaskInfo task: taskInfo) {
+            String taskName = task.getName();
+            String dueTime = this.viewModel.formatDeadline(task.getDeadline());
+            taskMap.put(taskName, dueTime);
+        }
+        return taskMap;
+    }
+
+
+    private Map<String, List<String>> formatEventInfos(List<EventInfo> eventInfos) {
+        Map<String, List<String>> eventMap = new HashMap<>();
+        List<String> list = new ArrayList<>();
+        if (eventInfos.isEmpty()){
+            eventMap.put("No Scheduled Events", null);
+        }
+
+        for (EventInfo event: eventInfos) {
+            String eventName = event.getName();
+
+            String startTime = event.getStartTime().format(
+                    DateTimeFormatter.ofLocalizedTime(
+                            FormatStyle.SHORT));
+
+            String endTime = event.getEndTime().format(
+                    DateTimeFormatter.ofLocalizedTime(
+                            FormatStyle.SHORT));
+            list.add(startTime);
+            list.add(endTime);
```

Comment:
How about just adding the `startTime + " - "  + endTime` here instead of up in line 100 when updating the view? That way the map can just be a `Map<String, String>` and you won't have to worry about doing the `get(0)` `get(1)`